cdap-client/src/main/java/co/cask/cdap/client/SecureStoreClient.java

```diff
@@ -0,0 +1,128 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.client;
+
+import co.cask.cdap.api.security.store.SecureStoreMetadata;
+import co.cask.cdap.client.config.ClientConfig;
+import co.cask.cdap.client.util.RESTClient;
+import co.cask.cdap.common.UnauthenticatedException;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.proto.id.SecureKeyId;
+import co.cask.cdap.proto.security.SecureKeyCreateRequest;
+import co.cask.cdap.proto.security.SecureKeyListEntry;
+import co.cask.common.http.HttpMethod;
+import co.cask.common.http.HttpResponse;
+import co.cask.common.http.ObjectResponse;
+import com.google.common.reflect.TypeToken;
+import com.google.gson.Gson;
+import com.google.inject.Inject;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.List;
+
+/**
+ * Provides ways to get/set Secure keys.
+ */
+public class SecureStoreClient {
+  private static final Gson GSON = new Gson();
+  private static final String SECURE_KEYS = "securekeys";
+
+  private final RESTClient restClient;
+  private final ClientConfig config;
+
+  @Inject
+  public SecureStoreClient(ClientConfig config, RESTClient restClient) {
+    this.config = config;
+    this.restClient = restClient;
+  }
+
+  public SecureStoreClient(ClientConfig config) {
+    this(config, new RESTClient(config));
+  }
+
+  /**
+   * Creates a secure key
+   *
+   * @param secureKeyId {@link SecureKeyId} secure key name
+   * @param keyCreateRequest {@link SecureKeyCreateRequest}
+   * @throws IOException if a network error occurred
+   * @throws UnauthenticatedException if the request is not authorized successfully in the gateway server
+   */
+  public void createKey(SecureKeyId secureKeyId,
+                        SecureKeyCreateRequest keyCreateRequest) throws IOException, UnauthenticatedException {
+    URL url = config.resolveNamespacedURLV3(secureKeyId.getParent().toId(), getSecureKeyPath(secureKeyId));
+    restClient.execute(HttpMethod.PUT, url, GSON.toJson(keyCreateRequest), null, config.getAccessToken());
+  }
+
+  /**
+   * Fetch the data associated with the given secure key
+   *
+   * @param secureKeyId {@link SecureKeyId} secure key name
+   * @return the secure data in a utf-8 encoded byte array
+   * @throws IOException if a network error occurred
+   * @throws UnauthenticatedException if the request is not authorized successfully in the gateway server
+   */
+  public String getData(SecureKeyId secureKeyId) throws IOException, UnauthenticatedException {
```

Comment:
I have opened a JIRA to address this https://issues.cask.co/browse/CDAP-6930. Will address it in a follow-up PR since it affects all other REST clients.